pkg/reconciler/utils/pubsub/subscription.go

```diff
@@ -56,26 +54,20 @@ func (r *Reconciler) ReconcileSubscription(ctx context.Context, id string, subCo
 			return nil, err
 		}
 		if config.Topic != nil && config.Topic.String() == deletedTopic {
-			logger.Error("The topic is deleted")
-			updater.MarkSubscriptionFailed("DeletedTopic", "Pull subscriptions must be recreated to work with recreated topic")
-			return nil, deletedTopicErr
+			logger.Warn("Detected deleted topic. Going to recreate the pull subscription.")
```

Comment:
Can we also record an Event (type Warn) saying that a deleted topic was detected? That way this situation is more clear from describing the resource. As written, the only indication from `kubectl describe` that something was amiss is a pair of Deleted Sub and Created Sub events (type Normal). A type Warn Event about the deleted topic is more clear that something exceptional has occurred.

Reply:
Good idea!